### ISSUE TITLE
[java] CloseResourceRule does not recognize multiple assignment done to resource

### ISSUE THREAD

**Issue by pbrajesh1** [pmd/pmd#2764]
**Affects PMD Version:**  PMD ??? (1.2.2 is an invalid version...)

**Rule:** [CloseResourceRule](https://pmd.github.io/latest/pmd_rules_java_errorprone.html#closeresource)

**Description:**
The CloseResourceRule does not identify in case the resource variable got re assigned by developer

**Code Sample demonstrating the issue:**

```java
public class Bar {
    public void withSQL() {
        Connection c = pool.getConnection();
        try {
            // do stuff
            // oops, should not have assgined the connection again to the variable 
            c = pool.getConnection();
        } catch (SQLException ex) {
           // handle exception
        } finally {
            // oops, one connection leak'!
            c.close();
        }
    }  
}
```

**Expected outcome:**

*   It is expected that PMD reports a violation of resource re assignment.  The other rules DataflowAnomalyAnalysisRule and java/bestpractices/UnusedAssignment is  not effective also to detect this .
*   Is PMD missing to report a violation, where there should be one? -> false-negative

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

**Comment by andipabst** [comment]
I'd be willing to work on this, if that's okay?
Is it okay to use DataFlowAnalysis for this?

**Comment by adangel** [comment]
> I'd be willing to work on this, if that's okay?
Sure, I've assigned it to you.

> Is it okay to use DataFlowAnalysis for this?

We should not use that anymore. It will be removed with PMD 7.

**Comment by andipabst** [comment]
Thanks!
Oh, I did miss that. Glad, that I asked and thanks for clarifying.
